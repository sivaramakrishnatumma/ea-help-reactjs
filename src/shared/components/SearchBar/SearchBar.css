.searchbar-container {
  position: relative;
}

.searchbar-container .fa-search {
  position: absolute;
  right: 12px;
  top: 11px;
  color: #ced4da;
}

.searchbar-input:focus + .fa-search, .searchbar-input:hover + .fa-search{
  display: none;
}