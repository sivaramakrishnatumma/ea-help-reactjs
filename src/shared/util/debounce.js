export const debounce = (fn, delay) => {
  let timer;
  return (...args) => {
    clearTimeout(timer);
    timer = setTimeout(() => {
      fn.apply(fn, args);
    }, delay);
  };
};
