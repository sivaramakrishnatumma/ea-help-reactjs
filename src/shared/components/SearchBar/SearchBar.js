import React from "react";
import "./SearchBar.css";
import { debounce } from "../../util/debounce";

const SearchBar = props => {
  const sendToProps = e => {
    props.onKeyPress(e.target.value);
  };

  const keyPressed = debounce(sendToProps, 300);

  return (
    <div className={"searchbar-container " + props.className}>
      <input
        className="form-control mr-sm-2 searchbar-input"
        type="search"
        placeholder={props.placeholder}
        aria-label="Search"
        onChange={e => {
          e.persist();
          keyPressed(e);
        }}
      />
      <i className="fa fa-search"></i>
    </div>
  );
};

export default SearchBar;
