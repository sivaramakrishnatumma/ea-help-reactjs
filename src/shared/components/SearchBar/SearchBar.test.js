import React from "react";
import SearchBar from "./SearchBar";
import { mount } from "enzyme";
import { debounce } from "../../util/debounce";

describe("SearchBar component test case", () => {
  it("should test SearchBar component with default props", () => {
    const props = {
      className: "search-bar",
      placeholder: "Search for Product",
      onKeyPress: jest.fn()
    };
    const keyPressed = jest.fn();
    const wrapper = mount(<SearchBar {...props} />);
    const event = { target: { value: "NewName" } };
    wrapper.find("input").simulate("change", event);
    expect(wrapper).toMatchSnapshot();
  });
});
