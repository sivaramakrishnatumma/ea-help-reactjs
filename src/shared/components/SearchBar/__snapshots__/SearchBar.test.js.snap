// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component test case should test SearchBar component with default props 1`] = `
<SearchBar
  className="search-bar"
  onKeyPress={[MockFunction]}
  placeholder="Search for Product"
>
  <div
    className="searchbar-container search-bar"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2 searchbar-input"
      onChange={[Function]}
      placeholder="Search for Product"
      type="search"
    />
    <i
      className="fa fa-search"
    />
  </div>
</SearchBar>
`;
